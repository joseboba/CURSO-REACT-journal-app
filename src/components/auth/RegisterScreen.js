import React from 'react'
import { useDispatch } from 'react-redux'
import { Link } from 'react-router-dom'
import validator from 'validator'
import { startRegisterWithEmailPasswordName } from '../../actions/actionsAuth'
import { removeError, setError } from '../../actions/actionsUi'
import { useForm } from '../../hooks/useForm'


export const RegisterScreen = () => {

    const dispatch = useDispatch();

    const [formValues, handleInputChange] = useForm({
       name: 'Jose Enrique', email: 'joseboba18@gmail.com', password: '123456', password2: '123456'
      })
    const { name, email, password, password2 } = formValues;


    const handleRegister = (e) => {
        e.preventDefault();
        if(isFormValid()){
          dispatch( startRegisterWithEmailPasswordName(email, password, name) ) 
        }
    }

    const isFormValid = () => {
        
        if(name.trim().length === 0){
          dispatch(setError('Name es requerido'))
          return false;
        }else if( !validator.isEmail(email) ){
          dispatch(setError('Email es incorrecto o no se ha escrito'))
          return false;
        }else if( password !== password2 || password.length < 5 ){
          dispatch(setError('La contraseña no es completamente segura o no coinciden'))
          return false;
        }
        dispatch(removeError())
        return true;
    }

    return (
        <>
        <h3 className="auth__title" > Register </h3>

        <form onSubmit={ handleRegister } autoComplete="off">

          <input
            type="text"
            placeholder="Name"
            name="name"
            autoComplete="off"
            className="auth_input"
            value={ name }
            onChange={ handleInputChange }
          />


          <input
            type="text"
            placeholder="Email"
            name="email"
            autoComplete="off"
            className="auth_input"
            value={ email }
            onChange={ handleInputChange }
          />

          <input
            type="password"
            placeholder="Password"
            name="password"
            autoComplete="off"
            className="auth_input"
            value={ password }
            onChange={ handleInputChange }
          />

          <input
            type="password"
            placeholder="Confirme su password"
            name="password2"
            autoComplete="off"
            className="auth_input"
            value={ password2 }
            onChange={ handleInputChange }
          />

          <button type="submit" className="btn btn-primary btn-block mt-5"  >Register</button>
        

            <Link 
                to="/auth/login"
                className="link "
            >
                ¿Ya estás registrado?
            </Link>

        </form>
      </>
    )
}
