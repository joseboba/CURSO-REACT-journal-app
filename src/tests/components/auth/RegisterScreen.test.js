import '@testing-library/jest-dom'
import React from 'react';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store'; //ES6 modules
import { mount } from 'enzyme'
import { RegisterScreen } from '../../../components/auth/RegisterScreen'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { types } from '../../../types/types';



const middlewares = [thunk]
const mockStore = configureStore(middlewares);
const initState = {};
let store = mockStore(initState)


describe('Pruebas en RegisterScreen', () => {

    const wrapper = mount(
        <Provider store={ store }>
            <MemoryRouter>
                <RegisterScreen />
            </MemoryRouter>
        </Provider>
    )

    test('debe de hacer el snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de hacer el dispatch de la acciÃ³n respectiva', () => {
        const emailField = wrapper.find('input[name="email"]')
        
        emailField.simulate('change', {
            target: {
                value: '',
                name: 'email'
            }
        })

        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        });
        const actions = store.getActions();
        
        expect( actions[0] ).toEqual({
            type: types.uiSetError,
            payload: 'Email es incorrecto o no se ha escrito'
        })

    })
    

})
