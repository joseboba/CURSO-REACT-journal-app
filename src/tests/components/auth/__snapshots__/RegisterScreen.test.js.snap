// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en RegisterScreen debe de hacer el snapshot 1`] = `
Array [
  <h3
    className="auth__title"
  >
     Register 
  </h3>,
  <form
    autoComplete="off"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth_input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Jose Enrique"
    />
    <input
      autoComplete="off"
      className="auth_input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="joseboba18@gmail.com"
    />
    <input
      autoComplete="off"
      className="auth_input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123456"
    />
    <input
      autoComplete="off"
      className="auth_input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirme su password"
      type="password"
      value="123456"
    />
    <button
      className="btn btn-primary btn-block mt-5"
      type="submit"
    >
      Register
    </button>
    <a
      className="link "
      href="/auth/login"
      onClick={[Function]}
    >
      ¿Ya estás registrado?
    </a>
  </form>,
]
`;
